### PR TITLE
Add non mediation pilot opt in app insights

### PR DESCRIPTION
**Pull request by dharmendrak**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-6928

### Change description ###

Add non mediation pilot opt in app insights

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessCallbackHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4545730&bid=15294441)

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessCallbackHandler.java [hunk 1]
```diff
@@ -63,9 +65,16 @@ public MediationSuccessCallbackHandler(
...
+            .trackEvent(getAppInsightEventBasedOnMediationPilot(claim), REFERENCE_NUMBER, claim.getReferenceNumber());
+
         return SubmittedCallbackResponse.builder().build();
     }
+
+    private AppInsightsEvent getAppInsightEventBasedOnMediationPilot(Claim claim) {
+        return FeaturesUtils.hasMediationPilotFeature(claim)
+            ? MEDIATION_PILOT_SUCCESS
+            : NON_MEDIATION_PILOT_SUCCESS;
+    }
```
Comment (daveoh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessCallbackHandler.java]: This looks like there is an amount of code duplication between the success and failure callback classes.

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessCallbackHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4545730&bid=15294441)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessCallbackHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4545730&bid=15294441)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4545730&bid=15294441)

**Pull request merged**